File: data/code_pairs/cpp/2163.cpp
// CommDataIO.hpp
#include <vector>

class CommDataIO {
public:
    static long long num_neib_domain_IO;
};

long long CommDataIO::num_neib_domain_IO = 0; // Definition

// CommunicationTable.hpp
#include <vector>

class CommunicationTable {
public:
    long long num_neib;

    // Constructor
    CommunicationTable() : num_neib(0) {}
};

// set_comm_table_4_IO.hpp
#include "CommDataIO.hpp"
#include "CommunicationTable.hpp"

void copy_comm_tbl_type_from_IO(CommunicationTable& comm_tbls) {
    comm_tbls.num_neib = CommDataIO::num_neib_domain_IO;
}

// main.cpp - Test Program
#include <iostream>
#include "set_comm_table_4_IO.hpp"

int main() {
    CommunicationTable comm_tbls;
    
    // Initialize with some test data
    CommDataIO::num_neib_domain_IO = 2;

    // Call the function
    copy_comm_tbl_type_from_IO(comm_tbls);

    // Verify the results
    std::cout << "C++ Test Output:" << std::endl;
    std::cout << "num_neib = " << comm_tbls.num_neib << std::endl;

    return 0;
}

File: data/code_pairs/fortran/2163.f90
! m_comm_data_IO.f90
module m_comm_data_IO
    use iso_fortran_env, only: int64
    implicit none
    integer(int64) :: num_neib_domain_IO = 0
end module m_comm_data_IO

! t_comm_table.f90
module t_comm_table
    use iso_fortran_env, only: int64
    implicit none

    type communication_table
        integer(int64) :: num_neib = 0
    end type communication_table
end module t_comm_table

! set_comm_table_4_IO.f90
module set_comm_table_4_IO
    use iso_fortran_env, only: int64
    use t_comm_table, only: communication_table
    use m_comm_data_IO, only: num_neib_domain_IO
    implicit none

contains

    subroutine copy_comm_tbl_type_from_IO(comm_tbls)
        type(communication_table), intent(inout) :: comm_tbls
        comm_tbls%num_neib = num_neib_domain_IO
    end subroutine copy_comm_tbl_type_from_IO
end module set_comm_table_4_IO

! test_comm_table.f90 - Test Program
program test_comm_table
    use set_comm_table_4_IO
    use m_comm_data_IO, only: num_neib_domain_IO
    implicit none

    type(communication_table) :: comm_tbl

    ! Initialize with some test data
    num_neib_domain_IO = 2

    ! Call the subroutine
    call copy_comm_tbl_type_from_IO(comm_tbl)

    ! Verify the results
    print *, "Fortran Test Output:"
    print *, "num_neib =", comm_tbl%num_neib
end program test_comm_table